# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/RootAccessible.cpp
# Commit: 06263018f1f7
# Full Hash: 06263018f1f70ceb780dff123c581c58c64d24e7
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-11-10 09:24:53
# Regressor Bug: 1733263
# File Overlap Count: 1
# Description:
#   Bug 1733263 - P5: Rely on DOMMenuItem events for ACTIVE state changes in select elements. r=Jamie
#   
#   1. Use dom events in RootAccessible to fire ACTIVE state changes.
#   2. Add DOMMenuItemInactive events to nsListControlFrame and fire it on
#      the previously active option.
# ==============================================================================

diff -r c8f226476a30 -r 06263018f1f7 accessible/generic/RootAccessible.cpp
--- a/accessible/generic/RootAccessible.cpp	Tue Nov 09 23:17:17 2021 +0000
+++ b/accessible/generic/RootAccessible.cpp	Tue Nov 09 23:17:17 2021 +0000
@@ -399,6 +399,9 @@
                               accessible);
     }
   } else if (eventType.EqualsLiteral("DOMMenuItemActive")) {
+    RefPtr<AccEvent> event =
+        new AccStateChangeEvent(accessible, states::ACTIVE, true);
+    nsEventShell::FireEvent(event);
     FocusMgr()->ActiveItemChanged(accessible);
 #ifdef A11Y_LOG
     if (logging::IsEnabled(logging::eFocus)) {
@@ -406,6 +409,10 @@
     }
 #endif
   } else if (eventType.EqualsLiteral("DOMMenuItemInactive")) {
+    RefPtr<AccEvent> event =
+        new AccStateChangeEvent(accessible, states::ACTIVE, false);
+    nsEventShell::FireEvent(event);
+
     // Process DOMMenuItemInactive event for autocomplete only because this is
     // unique widget that may acquire focus from autocomplete popup while popup
     // stays open and has no active item. In case of XUL tree autocomplete